app/main.py

```diff
@@ -1,3 +1,18 @@
-def cache(func):
-    # Write your code here
-    pass
+import functools
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cache_store = {}
+
+    @functools.wraps(func)
+    def wrapper(*args) -> Any:
+        if (func.__name__, args) in cache_store:
+            print("Getting from cache")
+            result = cache_store[(func.__name__, args)]
```

Comment:
As for me, it is redundant.
You can do return cache_store[(func.__name__, args)]
Also, why do you need to save the func name? 
<img width="435" alt="image" src="https://user-images.githubusercontent.com/65166203/216968833-4f8cf434-0947-42dd-a90c-898f41c54e61.png">

Reply:
<img width="209" alt="image" src="https://user-images.githubusercontent.com/65166203/216968924-dbc0544c-7134-4a3d-b88e-f81030bd5b7c.png">